framework/src/main/java/org/checkerframework/framework/type/treeannotator/PropagationTreeAnnotator.java

```diff
@@ -204,4 +209,27 @@ private boolean hasPrimaryAnnotationInAllHierarchies(AnnotatedTypeMirror type) {
         }
         return annotated;
     }
+
+    /**
+     * Adds the qualifiers in {@code annos} to {@code type} that are below the qualifier upper bound
+     * of type and for which type does not already have annotation in the same hierarchy. If a
+     * qualifier in {@code annos} is above the bound, then the bound is added to {@code type}
+     * instead.
+     *
+     * @param type annotations are added to this type
+     * @param annos annotations to add to type
+     */
+    private void addAnnoOrBound(AnnotatedTypeMirror type, Set<? extends AnnotationMirror> annos) {
```

Comment:
In other recent pull requests you used `quals` as the name for such qualifier sets.
Maybe `addQualOrBound` would also be a better method name?

Reply:
This file uses "annotation" rather "qualifier".  So `Anno` is more consistent.